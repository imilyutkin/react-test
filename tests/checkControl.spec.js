import React from 'react';
import {shallow} from 'enzyme';
import CheckboxControl from '../src/components/shared/CheckboxControl';
import renderer from 'react-test-renderer';

test('Checkbox test rendering', () => {
    const component = renderer.create(
        <CheckboxControl />
    );

    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();

    tree.props.onChange();
    tree = component.toJSON();
    expect(tree).toMatchSnapshot();
});

test('Checkbox test state changing', () => {
    const checkbox = shallow(
        <CheckboxControl />
    );

    expect(checkbox.find('input').prop('checked')).toBe(false);
    checkbox.find('input').simulate('change');
    expect(checkbox.find('input').prop('checked')).toBe(true);
});