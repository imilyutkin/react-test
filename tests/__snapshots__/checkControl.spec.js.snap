// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Checkbox test rendering 1`] = `
<input
  checked={false}
  onChange={[Function]}
  type="checkbox"
/>
`;

exports[`Checkbox test rendering 2`] = `
<input
  checked={true}
  onChange={[Function]}
  type="checkbox"
/>
`;
